core/src/main/kotlin/org/evomaster/core/problem/graphql/GraphQLActionBuilder.kt

```diff
@@ -208,85 +246,87 @@ object GraphQLActionBuilder {
+        val k = KindX()
+        val kindsLevels = k.quadKinds(elementInfields)
+
         tableElement.isKindOfTableFieldOptional = false
 
-        if (kind == LIST) {// non optional list
+        if (kindsLevels.kind1 == LIST) {// non optional list
             tableElement.kindOfTableField = LIST
 
-            if (kind2 == NON_NULL) {// non optional object or scalar or enum
+            if (kindsLevels.kind2 == NON_NULL) {// non optional object or scalar or enum or union or interface
                 tableElement.isKindOfTableFieldTypeOptional = false
-                tableElement.kindOfTableFieldType = kind3
-                tableElement.tableFieldType = elementInfields?.type?.ofType?.ofType?.ofType?.name
-                tableElement.tableType = elementIntypes?.name
+                tableElement.kindOfTableFieldType = kindsLevels.kind3
+                tableElement.tableFieldType = elementInfields.type.ofType.ofType.ofType.name
+                tableElement.tableType = elementIntypes.name
                 state.tables.add(tableElement)
-            } else {//optional object or scalar or enum
+            } else {//optional object or scalar or enum or union or interface
                 if (elementInfields?.type?.ofType?.ofType?.name == null) {
                     LoggingUtil.uniqueWarn(log, "Depth not supported yet ${elementIntypes}")
                 } else {
-                    tableElement.kindOfTableFieldType = kind2
+                    tableElement.kindOfTableFieldType = kindsLevels.kind2
                     tableElement.isKindOfTableFieldTypeOptional = true
-                    tableElement.tableFieldType = elementInfields?.type?.ofType?.ofType?.name
-                    tableElement.tableType = elementIntypes?.name
+                    tableElement.tableFieldType = elementInfields.type.ofType.ofType.name
+                    tableElement.tableType = elementIntypes.name
                     state.tables.add(tableElement)
                 }
             }
-        } else if (kind == OBJECT || kind == SCALAR || kind == ENUM) {
-            tableElement.kindOfTableFieldType = kind
-            tableElement.tableFieldType = elementInfields?.type?.ofType?.name
-            tableElement.tableType = elementIntypes?.name
+        } else if (kindsLevels.kind1?.let { kindChecks(it) }!!) {
+            tableElement.kindOfTableFieldType = kindsLevels.kind1
+            tableElement.tableFieldType = elementInfields.type.ofType.name
+            tableElement.tableType = elementIntypes.name
             state.tables.add(tableElement)
         } else {
-            LoggingUtil.uniqueWarn(log, "Type not supported yet:  ${elementInfields?.type?.ofType?.kind}")
+            LoggingUtil.uniqueWarn(log, "Type not supported yet:  ${elementInfields.type.ofType.kind}")
         }
 
     }
 
+    private fun kindChecks(kind: __TypeKind) = kind == OBJECT || kind == SCALAR || kind == ENUM || kind == UNION || kind == INTERFACE
```

Comment:
`kindChecks` is too ambiguous/non-descriptive.
also, when something returns a boolean, common practice (but not a rule) to start with `is`
what about something like `isObjectOrPrimitive` ?  
and add comment on the other kinds not in this list of check

Reply:
kind has a type: __TypeKind which is an enum class containing: SCALAR, OBJECT, INTERFACE, UNION, ENUM, INPUT_OBJECT, LIST, NON_NULL.
The repeated code was checking : 
 _**kind == INPUT_OBJECT || kind == SCALAR || kind == ENUM**_   or
 _**OBJECT || kind == SCALAR || kind == ENUM || kind == UNION || kind == INTERFACE**_
so we can not add others since we are checking exactly those.